src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -77,8 +79,32 @@ def update_data(self):
             iptuple = ip2tuple_v4(ip)
 
             subid = (if_index,) + iptuple
-            self.arp_dest_map[subid] = machex
-            self.arp_dest_list.append(subid)
+            arp_dest_map[subid] = machex
+            arp_dest_list.append(subid)
+        return arp_dest_map, arp_dest_list
+
+    def update_data(self):
+        self.arp_dest_map = {}
+        self.arp_dest_list = []
+        self.arp_dest_map, self.arp_dest_list = self.get_arp_table()
+        # For platform with multiple namespaces
+        # get arp table information from all namespace
+        if self.ns_list is not None:
+            # Get the current process namespace
+            current_fd = open(mibs.get_netns_path(nspid = os.getpid()))
+            for ns in self.ns_list:
+                dest_map = {}
+                dest_list = []
+                ns_fd = open(mibs.get_netns_path(nsname = ns))
+                # Change namespace temporarily
+                mibs.setns(ns_fd, mibs.CLONE_NEWNET)
+                dest_map, dest_list = self.get_arp_table()
+                self.arp_dest_map.update(dest_map)
+                self.arp_dest_list.extend(dest_list)
+                # Revert back to original namespace
+                mibs.setns(current_fd, mibs.CLONE_NEWNET)
```

Comment:
>mibs.setns(current_fd, mibs.CLONE_NEWNET) [](start = 16, length = 41)

Use a `finally` to make sure it is executed even there is exception?

Reply:
Modified implementation to use NEIGH_TABLE instead of using linux namespace.